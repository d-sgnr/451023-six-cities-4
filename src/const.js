export const RATING = {
  1: 20,
  2: 40,
  3: 60,
  4: 80,
  5: 100,
};

export const PropertyType = {
  CITY: `cities`,
  NEAR: `near-places`,
};

export const PLACE_TYPE = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`,
};

export const SortType = {
  POPULAR: `Popular`,
  TO_HIGH: `Price: low to high`,
  TO_LOW: `Price: high to low`,
  TOP_RATED: `Top rated first`,
};


export const PageType = {
  INDEX: `INDEX`,
  PROPERTY: `PROPERTY`,
};

export const Cities = {
  PARIS: {
    name: `Paris`,
    coordinates: [48.856663, 2.351556],
  },
  COLOGNE: {
    name: `Cologne`,
    coordinates: [50.930779, 6.938399],
  },
  BRUSSELS: {
    name: `Brussels`,
    coordinates: [50.851309, 4.351718],
  },
  AMSTERDAM: {
    name: `Amsterdam`,
    coordinates: [52.373057, 4.892557],
  },
  HAMBURG: {
    name: `Hamburg`,
    coordinates: [53.552645, 9.966287],
  },
  DUSSELDORF: {
    name: `Dusseldorf`,
    coordinates: [51.230569, 6.787428],
  },
};
