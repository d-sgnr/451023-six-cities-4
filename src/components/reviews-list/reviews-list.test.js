import React from "react";
import renderer from "react-test-renderer";
import ReviewsList from "./reviews-list.jsx";
import {RATING} from "../../const.js";

const reviews = [
  {
    id: Math.random(),
    avatar: `avatar-max.jpg`,
    userName: `Max`,
    text: `Am terminated it excellence invitation projection as. She graceful shy believed distance use nay. Lively is people so basket ladies window expect.`,
    rating: RATING[4],
    date: `2019-04-24`,
  },
  {
    id: Math.random(),
    avatar: `avatar-angelina.jpg`,
    userName: `Angelina`,
    text: `Though wished merits or be. Alone visit use these smart rooms ham. No waiting in on enjoyed placing it inquiry.`,
    rating: RATING[2],
    date: `2019-04-24`,
  },
];

it(`ReviewsList should be rendered correctly`, () => {
  const tree = renderer.create(
      <ReviewsList
        reviews={reviews}
      />).toJSON();

  expect(tree).toMatchSnapshot();
});
