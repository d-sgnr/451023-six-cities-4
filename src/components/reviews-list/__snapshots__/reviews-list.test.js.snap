// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsList should be rendered correctly 1`] = `
<ul
  className="reviews__list"
>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="avatar-max.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Max
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Am terminated it excellence invitation projection as. She graceful shy believed distance use nay. Lively is people so basket ladies window expect.
      </p>
      <time
        className="reviews__time"
        dateTime="2019-05-08T14:13:56.569Z"
      >
        April 2019
      </time>
    </div>
  </li>
  <li
    className="reviews__item"
  >
    <div
      className="reviews__user user"
    >
      <div
        className="reviews__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Reviews avatar"
          className="reviews__avatar user__avatar"
          height="54"
          src="avatar-anna.jpg"
          width="54"
        />
      </div>
      <span
        className="reviews__user-name"
      >
        Anna
      </span>
    </div>
    <div
      className="reviews__info"
    >
      <div
        className="reviews__rating rating"
      >
        <div
          className="reviews__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "30%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <p
        className="reviews__text"
      >
        Am terminated it excellence invitation projection as. She graceful shy believed distance use nay. Lively is people so basket ladies window expect.
      </p>
      <time
        className="reviews__time"
        dateTime="2019-05-08T14:13:56.569Z"
      >
        April 2019
      </time>
    </div>
  </li>
</ul>
`;
